### ISSUE TITLE
science: add Neural Quantum State baseline via NetKet — NB07

### ISSUE THREAD

**Issue by rosspeili** [ARPAQLS/spinq-vqe#9]
## Summary
Neural Quantum States (NQS) — variational wavefunctions parameterized by neural networks and optimized via Variational Monte Carlo — have been shown to match or exceed DMRG on the Kagome Heisenberg model (Astrakhantsev et al. 2021 PRX; Vieijra et al. 2021 PRL). Adding a NetKet NQS baseline:

1. Creates a four-way comparison on the same Hamiltonian: ED → VQE → NQS → DMRG
2. Directly addresses the "quantum advantage" question from reviewers
3. Significantly strengthens the paper for PRB/npj Quantum Materials

**NetKet** is JAX-native, pip-installable, and has built-in Heisenberg Hamiltonian support.

## Install

`pip install netket`

Add to `pyproject.toml`:

start code
[project.optional-dependencies]
nqs = ["netket"]
end code

## NB07 — `07_nqs_comparison.ipynb`

**NQS models (in order of complexity):**
1. RBM (Restricted Boltzmann Machine) — fast baseline
2. GCNN (Group-equivariant CNN) — respects Kagome lattice symmetries, state-of-the-art for frustrated magnets

**Content:**
1. Build Kagome graph in NetKet (convert from NetworkX adjacency)
2. Define Hilbert space (`nk.hilbert.Spin`) and Heisenberg Hamiltonian (`nk.operator.Heisenberg`)
3. Train RBM via VMC (Stochastic Reconfiguration optimizer), 300 iterations
4. Train GCNN via VMC for higher accuracy
5. Extended comparison table (extends Issue #8 table):

| N | ED E₀ | DMRG E₀ | NQS RBM | NQS GCNN | VQE E₀ |
|---|-------|---------|---------|---------|--------|
| 9 | −1.4219 | −1.4219 | ... | ... | −1.2846 |
| 12 | −1.4804 | ~−1.48 | ... | ... | −1.2035 |

6. Key figure: energy convergence per epoch for NQS vs VQE — shows relative efficiency
7. Export `data/method_comparison.csv`

## References
- Astrakhantsev et al. (2021) PRX 11, 041017 — NQS on Kagome Heisenberg
- Vieijra et al. (2021) PRL 126, 017102 — symmetry-equivariant NQS
- Carleo & Troyer (2017) Science 355 — original NQS paper

## Acceptance criteria
- RBM NQS converges within 5% of ED at N=9
- GCNN NQS achieves < 2% error at N=9 (matching literature)
- `data/method_comparison.csv` committed
- `netket` in `pyproject.toml [nqs]`
- **Depends on Issue #8 (TeNPy)** — DMRG baseline must exist first
